website/project/decorators.py

```diff
@@ -208,6 +208,9 @@ def check_can_access(node, user, key=None, api_node=None):
             if check_can_download_preprint_file(user, node):
                 return True
 
+        if node.is_deleted:
+            raise HTTPError(http.NOT_FOUND)
```

Comment:
Should this be a 410 instead of 404? Also, a custom message could help with debugging, e.g. `{'message_long': 'The node for this file has been deleted.'}`